Update trainers to Dalaran versions (Grand Master)

SoulSeekkor opened the pull request:
This updates the trainers to be those from Dalaran (with exception to cooking), so that they are neutral Grand Masters.  This also removes the need for alliance/horde specific trainers for enchanting, inscription, and jewelcrafting.

Included SQL that has to be ran manually (because of database names) but can be used to update existing creatures to the new ones using a simple find and replace and then executing it.

SoulSeekkor commented:
Also added new Stove object and updated position of various trainers and the anvil/forge.

Nelnamara commented:
Confirmed working.

Removing GH still does not clear DB, Possible issue with core.

SoulSeekkor commented:
Updated to remove the innkeeper as she serves literally no purpose at all.  Also updated so that enchanting, leatherworking, and alchemy actually have 2 trainers each.  This solves the problem of the Dalaran trainers not being "complete", they don't include all ranks or recipes unfortunately so a Master level trainer is needed to accompany them.

Also allow the ability to add the starter portals for each faction for free, this allows the module to be completely faction neutral now for trainers, portals, and auctioneers.
